### DISCUSSION TITLE
Roborock Q7 Max support

### DISCUSSION THREAD

**Discussion by flyman1664** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#273] [General]
Hi there,

Just to inform that roborock Q7 max seems to be supported (roborock.vacuum.a38).
I have one home. I could extract maps and can control device through HA. 

![image](https://user-images.githubusercontent.com/49916159/162686618-e79d5200-bdee-4ab0-93e9-2b30150c9cc9.png)

**Comment by cromelex** [comment]
Just to confirm this works for me as well, supporting multiple maps and room extracting.

I still get a warning regarding the roborock.vacuum.a38 not being supported - I suppose it should be added to the supported list. 
If you need any info to do so please let me know.

**Reply by hybrixxx** [reply]
For me it worked 1 day...... and it just stopped working. This is my config:
  - platform: xiaomi_cloud_map_extractor
    name: roborock_q7
    host: 192.168.1.97
    country: de
    token: !secret roborock_token_q7
    username: !secret xiaomi_cloud_username
    password: !secret xiaomi_cloud_password
    draw: ['all']
    attributes:
      - calibration_points
    map_transformation:
      rotate: 180

thx

**Reply by hybrixxx** [reply]
yesterday suddently started to work again and today stopped after some time....... weird....
